module4/homework/appendNewRecipe/AppendNewRecipe.cpp

```diff
@@ -0,0 +1,63 @@
+#include "AppendNewRecipe.hpp"
+#include <algorithm>
+#include <fstream>
+#include <map>
+
+bool AppendNewRecipe(std::vector<std::string> steps,
+                     const std::list<std::string>& ingredients,
+                     const std::deque<std::pair<size_t, char>>& amount) {
+    std::fstream recipes("recipes.txt", recipes.out | recipes.app);
+
+    if (!recipes.is_open())
+        return false;
+
+    std::stringstream recipiesSs = FormatRecipit(steps, ingredients, amount);
+    recipes << recipiesSs.str();
+    recipes.close();
+
+    return true;
+}
+
+std::vector<std::string> FormatIngredients(const std::list<std::string>& ingredients,
+                                           const std::deque<std::pair<size_t, char>>& amount) {
+    const std::map<char, std::string> amountString{{'g', "gram"},
+                                                   {'s', "szklanka(i)"},
+                                                   {'m', "mililitrow"}};
+
+    std::vector<std::string> formatedIngredientsVec;
+    formatedIngredientsVec.reserve(ingredients.size());
+    auto amountIt = amount.begin();
+    auto ingredientsIt = ingredients.begin();
+
+    std::transform(ingredients.begin(),
+                   ingredients.end(),
+                   amount.begin(),
+                   std::back_inserter(formatedIngredientsVec),
+                   [&amountString](const auto ingredient, const auto& value) {
+                       return (std::to_string(value.first) + ' ' + (amountString.find(value.second))->second + ' ' + ingredient);
```

Comment:
Maybe lets use std::stringstream. As far as I remember, Mateusz said that this code will run faster, because adding "strings" requires many operations

Reply:
+1